### PR TITLE
added tests for RandomForestClassifier class_weight and sample_weight & tests for KNN

### PR DESCRIPTION
**Pull request by OnlyDeniko**
Added tests which are testing accuracy ratio RandomForestClassifier between Scikit-learn and daal4py on the next parameters: sample_weight, class_weight.

### CODE REVIEW

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import pytest
+import random
+from sklearn.datasets import load_iris
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+from sklearn.ensemble import RandomForestClassifier as SKRandomForestClassifier
+from daal4py.sklearn.ensemble import RandomForestClassifier as D4PRandomForestClassifier
+
+def make_dataset(n_samples, n_features, random_state=37):
```
Comment (agorshk) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:26]: As I see, the function wasn't used in next part of file

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    D4P_model = D4PRandomForestClassifier(class_weight=weight)
+
+    SK_predict = SK_model.fit(X_train, y_train).predict(x_test)
+    D4P_predict = D4P_model.fit(X_train, y_train).predict(x_test)
+
+    SK_accuracy = accuracy_score(SK_predict, y_test)
+    D4P_accuracy = accuracy_score(D4P_predict, y_test)
+    ratio = D4P_accuracy / SK_accuracy
+
+    assert ratio >= 0.9
```
Comment (agorshk) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:63]: I suggest to add it as a parameter in function with default value equal to 0.9

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    D4P_model = D4PRandomForestClassifier()
+
+    SK_predict = SK_model.fit(X_train, y_train, sample_weight=weight[0]).predict(x_test)
+    D4P_predict = D4P_model.fit(X_train, y_train, sample_weight=weight[0]).predict(x_test)
+
+    SK_accuracy = accuracy_score(SK_predict, y_test)
+    D4P_accuracy = accuracy_score(D4P_predict, y_test)
+    ratio = D4P_accuracy / SK_accuracy
+
+    assert ratio >= 0.9, 'Failed testing sample weights, sample_weight_type = ' + weight[1]
```
Comment (agorshk) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:98]: the same, it's better to add this value as a parameter

File: daal4py/sklearn/neighbors/tests/test_kneighbors.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+DISTANCES = ['minkowski']
+ALGORITHMS = ['brute', 'kd_tree', 'auto']
+WEIGHTS = ['uniform', 'distance']
+KS = [1, 3, 7, 25]
+
+
+def make_dataset(n_samples=256, n_features=5, n_classes=2, random_state=777):
+    x, y = make_classification(n_samples=n_samples, n_features=n_features, n_classes=n_classes, random_state=random_state)
+    return train_test_split(x, y, random_state=777, test_size=0.5, shuffle=True)
```
Comment (agorshk) [inline daal4py/sklearn/neighbors/tests/test_kneighbors.py:33]: 1. random_state=random_state
2. may be it makes sense to add parameters test_size=0.5, shuffle=True to the function for future reuse it in different cases

Comment (OnlyDeniko) [reply daal4py/sklearn/neighbors/tests/test_kneighbors.py]: Corrected

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    D4P_accuracy = accuracy_score(D4P_predict, y_test)
+    ratio = D4P_accuracy / SK_accuracy
+
+    assert ratio >= check_ratio
+
+def make_filled_list(list_size, fill):
+    return [fill for i in range(list_size)]
+
+SAMPLE_WEIGHTS_IRIS = [
+    (make_filled_list(100, 0), '0'),
```
Comment (PetrovKP) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:65]:
```suggestion
    (np.full_like(range(100), 0), '0'),
```
try to use ready-made solutions

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    (make_filled_list(100, 1), '1'),
+    (make_filled_list(100, 5), '5'),
+    (make_filled_list(100, 50), '50'),
+    (make_filled_list(100, random.uniform(1, 50)), 'Random [1, 50]'),
+    (make_filled_list(100, random.uniform(1, 100)), 'Random [1, 100]'),
+    (make_filled_list(100, random.uniform(1, 500)), 'Random [1, 500]'),
+    (make_filled_list(100, random.uniform(1, 1000)), 'Random [1, 1000]'),
+    (make_filled_list(100, random.uniform(1, 5000)), 'Random [1, 5000]'),
+]
```
Comment (PetrovKP) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:74]: Are all weights the same here? What's the point? Why don't we look at different weight distributions?

**Review by PetrovKP — COMMENTED** [review]
We need to check every commit for these tests. for this it is necessary that these examples run during testing daal4py.

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+        D4P_model = D4PRandomForestClassifier()
+
+        SK_predict = SK_model.fit(X_train, y_train,
+                                  sample_weight=weight[0]).predict(x_test)
+        D4P_predict = D4P_model.fit(X_train, y_train,
+                                    sample_weight=weight[0]).predict(x_test)
+
+        SK_accuracy = accuracy_score(SK_predict, y_test)
+        D4P_accuracy = accuracy_score(D4P_predict, y_test)
+        ratio = D4P_accuracy / SK_accuracy
```
Comment (SmirnovEgorRu) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:138]: I suppose we need to check not only accuracy here by predicted labels, but additionally AUC/logloss by predicted probabilities. And probably, other metrics like F1, precision, ...

**Comment by PetrovKP** [comment]
@Mergifyio rebase

**Comment by OnlyDeniko** [comment]
@Mergifyio rebase

**Comment by PetrovKP** [comment]
@Mergifyio rebase

**Review by PetrovKP — APPROVED** [review]
